src/main/java/vn/triplet/controller/admin/UserController.java

```diff
@@ -1,14 +1,74 @@
 package vn.triplet.controller.admin;
 
+import java.util.List;
+
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import vn.triplet.model.User;
+import vn.triplet.model.User.Role;
+import vn.triplet.service.UserService;
 
 @Controller
 public class UserController {
 
+	private static final Logger logger = Logger.getLogger(UserController.class);
+	@Autowired
+	UserService userService;
+
 	@RequestMapping("/admin/users")
-	public String index() {
+	public String index(Model model) {
+		List<User> users = userService.loadUsers();
+		model.addAttribute("users", users);
 		return "views/admin/user/users";
 	}
 
+	@RequestMapping("/admin/users/{id}")
+	public String show(@PathVariable("id") int id, Model model) {
+		User user = userService.findById(id);
```

Comment:
Nếu trường hợp ko tìm thấy user  thì em vẫn hiển thì ra page `show` user ah, như vậy ko hợp lý, thường mình sẽ trả về page 404 not found hoặc page home với message ko tìm thấy user nhé

Reply:
Em đã sửa